hashes/src/lib.rs

```diff
@@ -39,14 +43,18 @@
 //! use std::io::Write as _;
 //! use bitcoin_hashes::Sha256;
 //!
+//! # fn hash() -> Result<Sha256, std::io::Error> {
 //! let part1: &[u8] = b"hello";
 //! let part2: &[u8] = b" ";
 //! let part3: &[u8] = b"world";
 //! let mut engine = Sha256::engine();
 //! engine.write_all(part1).expect("engine writes don't error");
-//! engine.write_all(part2).unwrap();
-//! engine.write_all(part3).unwrap();
```

Comment:
IIRC I wrote this, I used one `expect` and the rest `unwrap` so as not to repeat the message.

Reply:
I have changed it to all be `expect()` but it does now repeat.

Reply:
Ugh, why did you do that? This is a case of not being dogmatic about rules. Just because the API guidelines say no `unwrap` in this case the `unwrap` is warranted.

Reply:
Why? because I was blindly following the guidelines.  I have changed it back.  90% of the lines is still repetition so I thought of ways to reduce it, but I don't think 3 lines warrants a helper function, loop or anything.